moveit_ros/trajectory_cache/include/moveit/trajectory_cache/utils/utils.hpp

```diff
@@ -56,6 +58,21 @@ std::string getWorkspaceFrameId(const moveit::planning_interface::MoveGroupInter
 std::string getCartesianPathRequestFrameId(const moveit::planning_interface::MoveGroupInterface& move_group,
                                            const moveit_msgs::srv::GetCartesianPath::Request& path_request);
 
+/** @brief Restates a translation and rotation in a new frame.
+ *
+ * @param[in] tf. The transform buffer to use.
+ * @param[in] target_frame. The frame to restate in.
+ * @param[in] source_frame. The frame to restate from.
+ * @param[in,out] translation. The translation to restate. Ignored if nullptr.
+ * @param[in,out] rotation. The rotation to restate. Ignored if nullptr.
+ * @returns MoveItErrorCode::SUCCESS if successfully restated. Otherwise, will return return
+ * MoveItErrorCode::FRAME_TRANSFORM_FAILURE if the transform could not be retrieved.
+ */
+moveit::core::MoveItErrorCode restateInNewFrame(std::shared_ptr<tf2_ros::Buffer> tf, const std::string target_frame,
+                                                const std::string source_frame, geometry_msgs::msg::Point* translation,
+                                                geometry_msgs::msg::Quaternion* rotation,
```

Comment:
can this quaternion be a non-const reference instead of a pointer?

Reply:
I'm using a pointer for both message types to allow us to pass optional args. See the implementation of that method.

It was the only way to allow me to extract the function out, since the call sites sometimes only want to use one or the other.

I had considered making separate functions instead but that would lead to two TF buffer fetch calls instead of one, which would hurt performance at best, or cause TF desyncs at worst.